### PR TITLE
feat(app): Launch calibration wizards from Calibration Dashboard

### PR DESCRIPTION
**Pull request by jgbowser**
# Overview

Adds 3 new hooks that encapsulate the Wizards and launcher functions for Calibration Dashboard specific versions of the Deck, Tip Length, and Offset calibrations. The launcher functions are then passed to the useCalibrationTaskList hook so the ctas can be wired up to call the launcher functions onClick. Also touches up some logic in the TaskList component to ensure that the progress tracker on the left of the dashboard adheres to the designs at all steps of the calibration process

closes RAUT-292

## Demo

https://user-images.githubusercontent.com/66637570/211972881-2247f9cf-53a1-4f2b-9ed2-931b8ffe0a57.mov

# Changelog

- Added useDashboardCalibratePipOffset hook
- Added useDashboardCalibrateTipLength hook
- Added useDashboardCalibrateDeck hook
- Added poll in useDashboardTaskList hook to allow the task list to show most up-to-date calibration status
- Fixes some logic issues in the TaskList to ensure the progress tracker styling matches the designs for each possible status

# Review requests
Ensure the Calibration Dashboard shows the correct calibration statuses and launches the correct wizard when the CTA button is clicked. Ensure the calibrations complete successfully and the TaskList updates shortly after

# Risk assessment
Moderate, this is currently hidden behind feature flags, but changes how calibrations are completed which are vital to the robot functioning properly. Previous locations for calibrations should be untouched by these changes at this time though

### CODE REVIEW

File: app/src/organisms/Devices/hooks/useCalibrationTaskList.ts [hunk 1]
```diff
@@ -33,6 +53,16 @@ export function useCalibrationTaskList(robotName: string): TaskListProps {
   const tipLengthCalibrations = useTipLengthCalibrations(robotName)
   const offsetCalibrations = usePipetteOffsetCalibrations(robotName)
 
+  useInterval(
+    () => {
+      dispatch(fetchPipetteOffsetCalibrations(robotName))
+      dispatch(fetchTipLengthCalibrations(robotName))
+      dispatch(fetchCalibrationStatus(robotName))
+    },
+    5000,
```
Comment (b-cooper) [inline app/src/organisms/Devices/hooks/useCalibrationTaskList.ts]: 5000 seems like a good number. Conventionally, this should be stored in a CONSTANT_CASE `const` at the top of the file. e.g. `CALIBRATION_DATA_POLL_MS`

File: app/src/organisms/Devices/hooks/useCalibrationTaskList.ts [hunk 2]
```diff
@@ -55,11 +85,11 @@ export function useCalibrationTaskList(robotName: string): TaskListProps {
           })
         : ''
     // todo(jb, 2022-12-14): wire up ctas to actually launch wizards (RAUT-292)
-    deckTask.cta = { label: t('recalibrate'), onClick: () => {} }
+    deckTask.cta = { label: t('recalibrate'), onClick: () => deckCalLauncher() }
```
Comment (b-cooper) [inline app/src/organisms/Devices/hooks/useCalibrationTaskList.ts]:
```suggestion
    deckTask.cta = { label: t('recalibrate'), onClick: deckCalLauncher }
```

File: app/src/organisms/Devices/hooks/useCalibrationTaskList.ts [hunk 3]
```diff
@@ -55,11 +85,11 @@ export function useCalibrationTaskList(robotName: string): TaskListProps {
...
   } else {
     activeTaskIndices = [0, 0]
     deckTask.description = t('start_with_deck_calibration')
-    deckTask.cta = { label: t('calibrate'), onClick: () => {} }
+    deckTask.cta = { label: t('calibrate'), onClick: () => deckCalLauncher() }
```
Comment (b-cooper) [inline app/src/organisms/Devices/hooks/useCalibrationTaskList.ts]:
```suggestion
    deckTask.cta = { label: t('calibrate'), onClick: deckCalLauncher }
```

File: app/src/pages/Devices/CalibrationDashboard/hooks/useDashboardCalibratePipOffset.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  )
+
+  const [dispatchRequests] = RobotApi.useDispatchApiRequests(
+    dispatchedAction => {
+      if (
+        dispatchedAction.type === Sessions.ENSURE_SESSION &&
+        dispatchedAction.payload.sessionType ===
+          Sessions.SESSION_TYPE_PIPETTE_OFFSET_CALIBRATION
+      ) {
+        // @ts-expect-error(sa, 2021-05-26): avoiding src code change, use in operator to type narrow
```
Comment (b-cooper) [inline app/src/pages/Devices/CalibrationDashboard/hooks/useDashboardCalibratePipOffset.tsx]: these `@ts-expect-errors` can be removed if we assign them with this clunky pattern instead: 

```
'requestId' in dispatchedAction.meta ? dispatchedAction.meta.requestId ?? null : null
```

File: app/src/pages/Devices/CalibrationDashboard/hooks/useDashboardCalibrateTipLength.tsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+  const sessionType = Sessions.SESSION_TYPE_TIP_LENGTH_CALIBRATION
+  const withIntent = INTENT_TIP_LENGTH_OUTSIDE_PROTOCOL
+
+  const [dispatchRequests] = RobotApi.useDispatchApiRequests(
+    dispatchedAction => {
+      if (
+        dispatchedAction.type === Sessions.ENSURE_SESSION &&
+        dispatchedAction.payload.sessionType === sessionType
+      ) {
+        // @ts-expect-error TODO: account for possible absence of requestId on meta
```
Comment (b-cooper) [inline app/src/pages/Devices/CalibrationDashboard/hooks/useDashboardCalibrateTipLength.tsx]: same goes for these. `'requestId' in dispatchedAction.meta ? dispatchedAction.meta.requestId ?? null : null`

File: app/src/pages/Devices/CalibrationDashboard/hooks/useDashboardCalibratePipOffset.tsx [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+  PipetteOffsetCalibrationSessionParams,
+} from '../../../../redux/sessions/types'
+import type { RequestState } from '../../../../redux/robot-api/types'
+
+// pipette calibration commands for which the full page spinner should not appear
+const spinnerCommandBlockList: SessionCommandString[] = [
+  Sessions.sharedCalCommands.JOG,
+]
+
+const PIPETTE_OFFSET_TITLE = 'Pipette offset calibration'
```
Comment (jerader) [inline app/src/pages/Devices/CalibrationDashboard/hooks/useDashboardCalibratePipOffset.tsx]: Should these strings be in `i18n`?

File: app/src/organisms/Devices/hooks/useCalibrationTaskList.ts [hunk 7]
```diff
@@ -55,11 +87,11 @@ export function useCalibrationTaskList(robotName: string): TaskListProps {
           })
         : ''
     // todo(jb, 2022-12-14): wire up ctas to actually launch wizards (RAUT-292)
```
Comment (jerader) [inline app/src/organisms/Devices/hooks/useCalibrationTaskList.ts]: Can this `todo` be deleted now?

**Review by b-cooper — APPROVED** [review]
The code and its organization looks clean!  I tried it out and the CTA's are launching the flows as expected. The Deck Calibration's "creating session" spinner isn't working atm, but that can be addressed in a following PR.

**Review by jerader — APPROVED** [review]
Had a fun time going through the whole calibration dashboard in the simulator. Good job! 🦖 🌟 As Brian mentioned, I also noticed that the spinner isn't working.

File: app/src/organisms/Devices/hooks/useCalibrationTaskList.ts (hunk 7 above)
Comment (jgbowser) [reply app/src/organisms/Devices/hooks/useCalibrationTaskList.ts]: good catch

File: app/src/pages/Devices/CalibrationDashboard/hooks/useDashboardCalibratePipOffset.tsx (hunk 6 above)
Comment (jgbowser) [reply app/src/pages/Devices/CalibrationDashboard/hooks/useDashboardCalibratePipOffset.tsx]: ah you're right, I think I just pulled all this straight from another hook and didn't look too closely, I'll make this change real quick

**Comment by jgbowser** [comment]
> The code and its organization looks clean! I tried it out and the CTA's are launching the flows as expected. The Deck Calibration's "creating session" spinner isn't working atm, but that can be addressed in a following PR.

Ahh, okay, that becomes really obvious once I throttle the connection in the dev tools network tab. I'll go ahead and make that ticket and see if I can figure out what's going on

**Pull request merged**